libr/bin/format/mach0/mach0.c

```diff
@@ -381,6 +381,45 @@ static int parse_dysymtab(struct MACH0_(obj_t)* bin, ut64 off) {
 	return true;
 }
 
+static void parse_signature(struct MACH0_(obj_t) *bin, ut64 off) {
+    	int index, len;
+	ut32 count, data;
+	struct linkedit_data_command link = {};
+    	if (off > bin->size || off + sizeof(struct linkedit_data_command) > bin->size)
+	    	return;
+	len = r_buf_fread_at (bin->b, off, (ut8*)&link, bin->big_endian ? "4I" : "4i", 1);
+	if (len < 1) {
+		eprintf ("Failed to get data while parsing LC_CODE_SIGNATURE command\n");
+		return;
+	}
+	data =  link.dataoff;
+	if (data > bin->size || data + sizeof(struct super_blob_t) > bin->size)
+	    	return;
+	struct super_blob_t *super = (struct super_blob_t *) (bin->b->buf + data);
+	count = r_swap_ut32(super->count);
```

Comment:
Why do you cast some memory address to a struct and then swap endian?  Smells bad to me.